bytecode verification error related to hl.agg._prev_nonnull

to replicate:

```python
    def tmt2(table, row_key, col_key, row_fields=[], col_fields=[], n_partitions=None):
        ht = table.key_by()

        non_entry_fields = set(itertools.chain(row_key, col_key, row_fields, col_fields))
        entry_fields = [x for x in ht.row if x not in non_entry_fields]

        if not entry_fields:
            raise ValueError(...)

        col_data = hl.rbind(
            hl.array(
                hl.dict(
                    ht.aggregate(
                        hl.agg.collect_as_set(
                            (ht.row.select(*col_key),
                             ht.row.select(*col_fields)))))),
            lambda data: hl.struct(data=data,
                                   key_to_index=hl.dict(hl.range(0, hl.len(data)).map(lambda i: (data[i][0], i))))
        )

        col_data_uid = Env.get_uid()
        ht = ht.drop(*col_fields)
        ht = ht.annotate_globals(**{col_data_uid: col_data})

        entries_uid = Env.get_uid()
        ht = (ht.group_by(*row_key)
              .partition_hint(n_partitions)
              .aggregate(**{x: hl.agg._prev_nonnull(ht[x]) for x in row_fields},
                         **{entries_uid: hl.rbind(
                             hl.dict(hl.agg.collect((ht[col_data_uid]['key_to_index'][ht.row.select(*col_key)],
                                                     ht.row.select(*entry_fields)))),
                             lambda entry_dict: hl.range(0, hl.len(ht[col_data_uid]['key_to_index']))
                                 .map(lambda i: entry_dict.get(i)))
                         }))
        ht = ht.annotate_globals(
            **{col_data_uid: hl.array(ht[col_data_uid]['data'].map(lambda elt: hl.struct(**elt[0], **elt[1])))})
        return ht._unlocalize_entries(entries_uid, col_data_uid, col_key)

N, M = 50, 50
mt = hl.utils.range_matrix_table(N, M)
mt = mt.key_cols_by(s = 'Col' + hl.str(M - mt.col_idx))
mt = mt.annotate_cols(c1 = hl.rand_bool(0.5))
mt = mt.annotate_rows(r1 = hl.rand_bool(0.5))
mt = mt.annotate_entries(e1 = hl.rand_bool(0.5))

re_mt = tmt2(mt.entries(), ['row_idx'], ['s'], ['r1'], ['col_idx', 'c1'])
new_col_order = re_mt.col_idx.collect()
mapping = [t[1] for t in sorted([(old, new) for new, old in enumerate(new_col_order)])]

assert re_mt.choose_cols(mapping).drop('col_idx')._same(mt.drop('col_idx'))
```

```
Caused by: is.hail.relocated.org.objectweb.asm.tree.analysis.AnalyzerException: Argument 2: expected J, but found I
	at is.hail.relocated.org.objectweb.asm.tree.analysis.BasicVerifier.naryOperation(Unknown Source)
	at is.hail.relocated.org.objectweb.asm.tree.analysis.BasicVerifier.naryOperation(Unknown Source)
	at is.hail.relocated.org.objectweb.asm.tree.analysis.Frame.execute(Unknown Source)
	... 35 more
method1(Lis/hail/annotations/Region;[Lis/hail/annotations/aggregators/RegionValueAggregator;JZJZ)V
00000 C118 Region RegionValueAggregator J . I J . I . . .  :  :     ILOAD 8
00001 C118 Region RegionValueAggregator J . I J . I . . .  : I  :     ISTORE 9
00002 C118 Region RegionValueAggregator J . I J . I I . .  :  :     ILOAD 9
00003 C118 Region RegionValueAggregator J . I J . I I . .  : I  :     IFEQ L0
00004 C118 Region RegionValueAggregator J . I J . I I . .  :  :     GOTO L1
00005 C118 Region RegionValueAggregator J . I J . I I . .  :  :    L0
00006 C118 Region RegionValueAggregator J . I J . I I . .  :  :     LLOAD 6
00007 C118 Region RegionValueAggregator J . I J . I I . .  : J  :     GOTO L2
00008 ?                      :    L1
00009 ?                      :     LDC 0
00010 C118 Region RegionValueAggregator J . I J . I I . .  : J  :    L2
00011 C118 Region RegionValueAggregator J . I J . I I . .  : J  :     LSTORE 10
00012 C118 Region RegionValueAggregator J . I J . I I J .  :  :     ALOAD 0
00013 C118 Region RegionValueAggregator J . I J . I I J .  : C118  :     ILOAD 9
00014 C118 Region RegionValueAggregator J . I J . I I J .  : C118 I  :     IFEQ L3
00015 C118 Region RegionValueAggregator J . I J . I I J .  : C118  :     GOTO L4
00016 C118 Region RegionValueAggregator J . I J . I I J .  : C118  :    L3
00017 C118 Region RegionValueAggregator J . I J . I I J .  : C118  :     ALOAD 1
00018 C118 Region RegionValueAggregator J . I J . I I J .  : C118 Region  :     LLOAD 10
...
```

--- a/hail/python/hail/expr/aggregators/aggregators.py
+++ b/hail/python/hail/expr/aggregators/aggregators.py
@@ -1418,7 +1418,13 @@ def group_by(group, agg_expr) -> DictExpression:
 
 @typecheck(expr=expr_any)
 def _prev_nonnull(expr) -> ArrayExpression:
-    return _agg_func('PrevNonnull', [expr], expr.dtype, [])
+    wrap = expr.dtype in {tint32, tint64, tfloat32, tfloat64, tbool, tcall}
+    if wrap:
+        expr = hl.tuple([expr])
+    r = _agg_func('PrevNonnull', [expr], expr.dtype, [])
+    if wrap:
+        r = r[0]
+    return r
 
 @typecheck(f=func_spec(1, expr_any),
            array=expr_array())

--- a/hail/python/test/hail/expr/test_expr.py
+++ b/hail/python/test/hail/expr/test_expr.py
@@ -2516,3 +2516,8 @@ class Tests(unittest.TestCase):
 
         with pytest.raises(hl.utils.FatalError):
             hl.eval(hl.bit_rshift(hl.int64(1), -1, logical=True))
+
+    def test_prev_non_null(self):
+        ht = hl.utils.range_table(1)
+
+        assert ht.aggregate((hl.agg._prev_nonnull(ht.idx))) == 0

